packages/design-system/src/components/UsaBanner/UsaBanner.jsx

```diff
@@ -10,14 +17,89 @@ export class UsaBanner extends React.PureComponent {
   constructor(props) {
     super(props);
     this.id = props.id || uniqueId('gov-banner_');
-    this.state = { isBannerOpen: false };
+    this.media = window.matchMedia('(max-width: 543px)');
+    this.state = {
+      isBannerOpen: false,
+      shouldRenderMobileView: this.media.matches,
+    };
     this.handleToggleBanner = this.handleToggleBanner.bind(this);
+    this.onMediaChange = this.onMediaChange.bind(this);
+
+    this.media.addEventListener('change', this.onMediaChange);
```

Comment:
Does this need to be cleaned up when the component goes away? I guess since this is in the header, that shouldn't really happen, should it?

Reply:
I'll add a cleanup function in here for safety! I think generally we'd be okay since the component is usually not mounting and un-mounting on a page, but also, it feels short sited to assume that.